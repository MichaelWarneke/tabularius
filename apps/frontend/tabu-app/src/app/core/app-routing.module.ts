import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

const routes: Routes = [
  {
    path: 'dash',
    loadChildren: '@tabularius/modules/app-dash#AppDashModule'
  },
  {
    path: 'journal',
    loadChildren: '@tabularius/modules/app-journal#AppJournalModule'
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: []
})
export class AppRoutingModule {}
