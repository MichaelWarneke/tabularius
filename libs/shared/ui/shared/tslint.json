{
  "extends": "../../../../tslint.json",
  "rules": {
    "directive-selector": [true, "attribute", "tabuSharedUi", "camelCase"],
    "component-selector": [true, "element", "tabu-shared-ui", "kebab-case"]
  }
}
