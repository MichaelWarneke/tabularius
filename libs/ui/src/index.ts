export { TabuAppSidenavModule } from './lib/nav';
