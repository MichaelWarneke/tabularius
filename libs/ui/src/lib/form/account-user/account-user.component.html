<h1>Account Form</h1>
<tabu-dynamic-form
  [formModel]="formModel"
  [data]="user"
  (saveEntry)="onSave($event)"
></tabu-dynamic-form>
