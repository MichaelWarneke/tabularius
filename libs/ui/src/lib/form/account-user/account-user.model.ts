import { IUserAbstract } from '@tabularius/shared/models';
import {
  FormControlTextbox,
  FormControlDate,
  Validators
} from '@tabularius/shared/dynamic-form';

export class UserFormModel implements IUserAbstract {
  uid = new FormControlTextbox('', [Validators.required]);
  email = new FormControlTextbox('', [Validators.required]);
  displayName = new FormControlTextbox('', [Validators.required]);
  photoURL = new FormControlTextbox('', []);
  constructor() {
    this.uid.label = 'User ID';
    this.email.label = 'Email';
    this.displayName.label = 'User Name';
    this.photoURL.label = 'Photo URL';
  }
}

export class FormSetup {
  model = new UserFormModel();
  submitButtonText = 'Save';
}
