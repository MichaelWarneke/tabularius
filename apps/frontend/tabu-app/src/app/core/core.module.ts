import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';

import { AppRoutingModule } from './app-routing.module';
import { TabuAppSidenavModule } from '@tabularius/ui';

@NgModule({
  declarations: [],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    TabuAppSidenavModule
  ],
  exports: [TabuAppSidenavModule]
})
export class CoreModule {}
