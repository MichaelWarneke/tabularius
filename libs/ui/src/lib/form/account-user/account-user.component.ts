import {
  Component,
  OnInit,
  Input,
  Output,
  EventEmitter,
  ChangeDetectionStrategy
} from '@angular/core';
import { IUser } from '@tabularius/shared/models';
import { FormSetup } from './account-user.model';

@Component({
  selector: 'tabu-account-user',
  templateUrl: './account-user.component.html',
  styleUrls: ['./account-user.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  providers: []
})
export class AccountUserComponent implements OnInit {
  _user: IUser | null = null;
  formModel = new FormSetup();
  @Output() save = new EventEmitter<IUser>();

  @Input()
  set user(user: IUser | null) {
    this._user = user;
  }

  get user() {
    return this._user;
  }

  constructor() {}

  ngOnInit() {}

  onSave(data: any) {
    console.warn('onSave :', data);
    if (data) {
      this.save.emit({
        ...data
      } as IUser);
    }
  }


}
