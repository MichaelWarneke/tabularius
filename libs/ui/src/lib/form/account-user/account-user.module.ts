import { NgModule } from '@angular/core';
import { AccountUserComponent } from './account-user.component';
import { SharedModule } from '@tabularius/shared/ui';
import { DynamicFormModule } from '@tabularius/shared/dynamic-form';

@NgModule({
  declarations: [AccountUserComponent],
  imports: [SharedModule, DynamicFormModule],
  exports: [AccountUserComponent]
})
export class AccountUserModule {}
