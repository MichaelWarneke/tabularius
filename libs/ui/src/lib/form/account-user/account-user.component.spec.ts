import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { AccountUserComponent } from './account-user.component';
import { SharedUiModule } from '@tabularius/shared/ui';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

describe('AccountUserComponent', () => {
  let component: AccountUserComponent;
  let fixture: ComponentFixture<AccountUserComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [NoopAnimationsModule, SharedUiModule],
      declarations: [AccountUserComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AccountUserComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
