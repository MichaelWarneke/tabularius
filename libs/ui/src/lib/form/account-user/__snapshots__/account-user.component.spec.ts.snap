// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountUserComponent should render 1`] = `
<tabu-account-user
  _user="null"
  formModel={[Function FormSetup]}
  save={[Function EventEmitter]}
>
  <h1>
    Account Form
  </h1><tabu-dynamic-form />
</tabu-account-user>
`;
